yt_dlp/extractor/tiktok.py

```diff
@@ -793,102 +792,150 @@ def _real_extract(self, url):
+            'webcast_language': 'en',
         }
 
+    def _entries(self, sec_uid, user_name):
+        cursor = int(time.time() * 1E3)
         for page in itertools.count(1):
-            for retry in self.RetryManager():
-                try:
-                    post_list = self._call_api(
-                        'aweme/post', query, username, note=f'Downloading user video list page {page}',
-                        errnote='Unable to download user video list')
-                except ExtractorError as e:
-                    if isinstance(e.cause, json.JSONDecodeError) and e.cause.pos == 0:
-                        retry.error = e
-                        continue
-                    raise
-            yield from post_list.get('aweme_list', [])
-            if not post_list.get('has_more'):
+            response = self._download_json(
+                self._API_BASE_URL, user_name or sec_uid, f'Downloading page {page}',
+                query=self._build_web_query(sec_uid, cursor), headers={'User-Agent': self._USER_AGENT})
+
+            for video in traverse_obj(response, ('itemList', lambda _, v: v['id'])):
+                video_id = video['id']
+                webpage_url = self._create_url(user_name, video_id)
+                info = try_call(
+                    lambda: self._parse_aweme_video_web(video, webpage_url, video_id)) or {'id': video_id}
+                info.pop('formats', None)
+                yield self.url_result(webpage_url, TikTokIE, **info)
+
+            old_cursor = cursor
+            cursor = traverse_obj(
+                response, ('itemList', -1, 'createTime', {lambda x: x * 1E3}, {int_or_none}))
+            if not cursor:
+                cursor = old_cursor - 604800000  # jump 1 week back in time
+            if cursor < 1472706000000 or not traverse_obj(response, 'hasMorePrevious'):
                 break
-            query['max_cursor'] = post_list['max_cursor']
-
-    def _entries_api(self, user_id, videos):
-        for video in videos:
-            yield {
-                **self._parse_aweme_video_app(video),
-                'extractor_key': TikTokIE.ie_key(),
-                'extractor': 'TikTok',
-                'webpage_url': f'https://tiktok.com/@{user_id}/video/{video["aweme_id"]}',
-            }
+
+    def _get_sec_uid(self, user_url, user_name, msg):
+        webpage = self._download_webpage(
+            user_url, user_name, fatal=False, headers={'User-Agent': 'Mozilla/5.0'},
+            note=f'Downloading {msg} webpage', errnote=f'Unable to download {msg} webpage') or ''
+        return traverse_obj(
+            self._get_universal_data(webpage, user_name),
+            ('webapp.user-detail', 'userInfo', 'user', 'secUid', {str})) or traverse_obj(
+            self._get_sigi_state(webpage, user_name),
+            ('LiveRoom', 'liveRoomUserInfo', 'user', 'secUid'),
+            ('UserModule', 'users', ..., 'secUid'),
+            get_all=False, expected_type=str)
```

Comment:
Personally, I try to avoid this type of intending where it's not obvious where one traverse_obj ends and next begins. But not a big deal
```suggestion
        return (traverse_obj(self._get_universal_data(webpage, user_name),
                             ('webapp.user-detail', 'userInfo', 'user', 'secUid', {str}))
                or traverse_obj(self._get_sigi_state(webpage, user_name),
                                ('LiveRoom', 'liveRoomUserInfo', 'user', 'secUid', {str}),
                                ('UserModule', 'users', ..., 'secUid', {str}, {any})))
```